File: Algorithm/Data_Structure/Bit_Operations/2595_easy.go
package Bit_Operations

/*
    @Author  : https://github.com/hakusai22
    @Time    : 2024/10/01 23:30
    @题目     : https://leetcode.cn/problems/number-of-even-and-odd-bits/description/
    @参考     :
    @时间复杂度:
    @空间复杂度:

 数据范围:

*/

func evenOddBit(n int) []int {
	ans := make([]int, 2)
	for i := 0; n > 0; i++ {
		if n&1 == 1 {
			ans[i%2]++
		}
		n >>= 1
	}
	return ans
}

File: Algorithm/Data_Structure/Bit_Operations/2595_easy.js
/**
 * @Time 2024/10/01 23:30
 * @Author  : https://github.com/hakusai22
 */

/**
 * @param {number} n
 * @return {number[]}
 */
var evenOddBit = function (n) {
    let ans = [0, 0];
    let i = 0;
    while (n) {
        if (n & 1) ans[i % 2]++;
        i++;
        n >>= 1;
    }
    return ans;
};